python/dataset.cpp

```diff
@@ -240,25 +240,42 @@ void init_dataset(py::module &m) {
 
   py::class_<DataArray> dataArray(m, "DataArray", R"(
     Named variable with associated coords, masks, and attributes.)");
-  dataArray.def(py::init<const DataArrayConstView &>());
+  py::options options;
+  options.disable_function_signatures();
   dataArray
-      .def(py::init([](VariableConstView data,
-                       std::map<Dim, VariableConstView> coords,
-                       std::map<std::string, VariableConstView> masks,
-                       std::map<Dim, VariableConstView> unaligned_coords,
-                       const std::string &name) {
-             return DataArray{Variable{data}, coords, masks, unaligned_coords,
-                              name};
-           }),
-           py::arg("data") = Variable{},
-           py::arg("coords") = std::map<Dim, VariableConstView>{},
-           py::arg("masks") = std::map<std::string, VariableConstView>{},
-           py::arg("attrs") = std::map<Dim, VariableConstView>{},
-           py::arg("name") = std::string{})
+      .def(
+          py::init([](VariableConstView data,
+                      std::map<Dim, VariableConstView> coords,
+                      std::map<std::string, VariableConstView> masks,
+                      std::map<Dim, VariableConstView> unaligned_coords,
+                      const std::string &name) {
+            return DataArray{Variable{data}, coords, masks, unaligned_coords,
+                             name};
+          }),
+          py::arg("data") = Variable{},
+          py::arg("coords") = std::map<Dim, VariableConstView>{},
+          py::arg("masks") = std::map<std::string, VariableConstView>{},
+          py::arg("attrs") = std::map<Dim, VariableConstView>{},
+          py::arg("name") = std::string{},
+          R"(__init__(self, data: VariableConstView, coords: Dict[str, VariableConstView] = {}, masks: Dict[str, VariableConstView] = {}, attrs: Dict[str, VariableConstView] = {}, name: str = '') -> None
```

Comment:
Views (or their names) are not explicitly exposed to Python users, so we should not refer to `VariableConstView`, but `Variable` (even if it actually accepts a `VariableConstView`).

Reply:
I agree that it is better to write 'Variable' here.
But the views *are* bound to Python and appear in some of scipp's output, however they are not documented. That is its own issue, though.

Reply:
Yes, the plan is to avoid exposing the view names to the user, see, e.g., https://github.com/scipp/scipp/issues/1302 with a suggestion to remove it from auto-complete (and others).